### ISSUE TITLE
[rds]: specify a name for the secret will destroy the old cluster and create a new one

### ISSUE THREAD

**Issue by sunshineo** [aws/aws-cdk#13891]
After specify the name of the secret that saves the admin password, the deploy cause the cluster to be destroyed and then create a new one.

### Reproduction Steps
1. Create an RDS cluster without specify the name for the secret and deploy. A secret for the admin password is created with a generated name like `DatabaseSecret3B817195-CjKSXGL6fXiP`
```typescript
const vpc = new ec2.Vpc(stack, 'VPC')
new rds.DatabaseCluster(stack, 'Database', {
    engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_08_1 }),
    instances: 3,
    instanceProps: {
      vpcSubnets: {
        subnetType: ec2.SubnetType.PUBLIC,
      },
      vpc,
    },
  })
```
2. Update the code to specify a name for the secret and deploy.
```typescript
const vpc = new ec2.Vpc(stack, 'VPC')
new rds.DatabaseCluster(stack, 'Database', {
    engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_08_1 }),
    instances: 3,
    instanceProps: {
      vpcSubnets: {
        subnetType: ec2.SubnetType.PUBLIC,
      },
      vpc,
    },
+    credentials: rds.Credentials.fromGeneratedSecret('admin', {
+      secretName: 'my-secret-name',
+    }),
  })
```

### What did you expect to happen?
The secret name changed from `DatabaseSecret3B817195-CjKSXGL6fXiP` to the `my-secret-name`. Or `DatabaseSecret3B817195-CjKSXGL6fXiP` was deleted and `my-secret-name` was created with the same content. The cluster is kept unchanged.

### What actually happened?
The cluster was destroyed and a new cluster was created

### Environment

  - **CDK CLI Version  :**1.92.0
  - **Framework Version:**1.95.1
  - **Node.js Version:**  v14.13.0
  - **OS               :**Mac
  - **Language (Version):** TypeScript (3.9.7)

### Other

--- 

This is :bug: Bug Report

**Comment by skinny85** [comment]
You're correct @sunshineo . That was made to preserve backwards compatibility. You can use the `fromUsername()` method instead of `fromGeneratedSecret()` which preserves this legacy behavior.

Changing this default would have to de done behind a feature flag, like [this change](https://github.com/aws/aws-cdk/pull/13710), to preserve backwards compatibility.

**Comment by sunshineo** [comment]
@skinny85 But using `fromUsername` gives a deprecated warning. So at some point we will not be able to move forward the cdk version?

**Comment by skinny85** [comment]
@sunshineo yes, it does, because we don't recommend using that method going forward - using the non-deprecated methods will avoid this problem altogether.

@jogold what do you think? Should we un-deprecate these methods, to aid people with migrations?

**Comment by jogold** [comment]
> @jogold what do you think? Should we un-deprecate these methods, to aid people with migrations?

I'm not sure. Even if you un-deprecate the methods, there's no way to rename the secret here without replacing the cluster, no?

**Comment by sunshineo** [comment]
Is this fixed? Reading this after 1 year makes me wonder, how this bug preserves backward compatibility?
